1.13 Bug Report Thread

ESPHome v1.13 is currently in beta phase: [beta.esphome.io](https://www.youtube.com/watch?v=dQw4w9WgXcQ) [🎉](https://beta.esphome.io)

The version also brings a big refactor on the C++ generator side. However, that also means some components could have stopped working.

This thread is for quickly finding and reporting issues in the current beta release so that the 1.13 release can be as stable as possible. You can post it as a new issue, or use this thread with the form provided below (this thread is here so that creating a bug report is quicker)

Most errors will be on the "this config does not compile" side. I don't expect any errors that would limit an ESPs ability to roll back if there's an issue.

If you find an issue, please do report it! Thanks!

````md
Component: <COMPONENT THAT HAS ISSUE>
Description:
DESCRIPTION OF PROBLEM

Configuration:
```yaml
# PASTE YAML CONFIG HERE

```

Additional Info:

````

<!-- LEAVE THIS LINE AS-IS AND DON'T DELETE IT, OTHERWISE THE ISSUE WILL BE CLOSED AUTOMATICALLY. -->

Component: Platform io compilation issue on hassio esphome 1.13
Description:
See build log below config.

Configuration:
```yaml
esphome:
  name: werkbank
  platform: ESP32
  board: nodemcu-32s

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

# Enable logging
logger:

# Enable Home Assistant API
api:
  password: !secret esphome_password

ota:
  password: !secret esphome_password

output:
  - platform: ledc
    pin: 17
    id: gpio_17_red
  - platform: ledc
    pin: 5
    id: gpio_5_green
  - platform: ledc
    pin: 18
    id: gpio_18_blue
  - platform: ledc
    pin: 19
    id: gpio_19_white

light:
  - platform: rgbw
    name: "Werkbank"
    red: gpio_17_red
    green: gpio_5_green
    blue: gpio_18_blue
    white: gpio_19_white
```

Additional Info:
```
INFO Reading configuration...
INFO Generating C++ source...
INFO Compiling app...
INFO Running:  platformio run -d /config/esphome/werkbank
Processing werkbank (platform: espressif32@1.6.0; framework: arduino; board: nodemcu-32s)
--------------------------------------------------------------------------------
LibraryManager: Installing id=1826 @ 1.0.3
Downloading...
Unpacking  [####################################]  100%
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/nodemcu-32s.html
PLATFORM: Espressif 32 > NodeMCU-32S
HARDWARE: ESP32 240MHz 320KB RAM (4MB Flash)
DEBUG: CURRENT(esp-prog) EXTERNAL(esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
Library Dependency Finder -> http://bit.ly/configure-pio-ldf
LDF MODES: FINDER(chain) COMPATIBILITY(soft)
Collected 27 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <AsyncTCP> 1.0.3
|-- <ESPmDNS> 1.0
|   |-- <WiFi> 1.0
|-- <Preferences> 1.0
|-- <WiFi> 1.0
|-- <Update> 1.0
Compiling /data/werkbank/.pioenvs/werkbank/src/esphome/components/api/api_message.cpp.o
Compiling /data/werkbank/.pioenvs/werkbank/src/esphome/components/api/api_server.cpp.o
Compiling /data/werkbank/.pioenvs/werkbank/src/esphome/components/api/basic_messages.cpp.o
Compiling /data/werkbank/.pioenvs/werkbank/src/esphome/components/api/command_messages.cpp.o
In file included from /root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/tuple:40:0,
from /root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/functional:55,
from src/esphome/core/component.h:4,
from src/esphome/components/api/api_message.h:3,
from src/esphome/components/api/api_message.cpp:1:
/root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/bits/uses_allocator.h:104:2: error: invalid preprocessing directive #en
#en
^
/root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/bits/uses_allocator.h:25:0: error: unterminated #ifndef
#ifndef _USES_ALLOCATOR_H
^
In file included from /root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/memory:77:0,
from src/esphome/core/helpers.h:6,
from src/esphome/components/api/util.h:3,
from src/esphome/components/api/api_message.h:4,
from src/esphome/components/api/api_message.cpp:1:
/root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/bits/uses_allocator.h:25:0: error: unterminated #ifndef
#ifndef _USES_ALLOCATOR_H
^
In file included from /root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/tuple:40:0,
from /root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/functional:55,
from src/esphome/core/component.h:4,
from src/esphome/components/api/api_server.h:3,
from src/esphome/components/api/api_server.cpp:3:
/root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/bits/uses_allocator.h:104:2: error: invalid preprocessing directive #en
#en
^
/root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/bits/uses_allocator.h:25:0: error: unterminated #ifndef
#ifndef _USES_ALLOCATOR_H
^
In file included from /root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/tuple:40:0,
from /root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/functional:55,
from src/esphome/core/component.h:4,
from src/esphome/components/api/command_messages.h:3,
from src/esphome/components/api/command_messages.cpp:1:
/root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/bits/uses_allocator.h:104:2: error: invalid preprocessing directive #en
#en
^
/root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/bits/uses_allocator.h:25:0: error: unterminated #ifndef
#ifndef _USES_ALLOCATOR_H
^
In file included from /root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/tuple:40:0,
from /root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/functional:55,
from src/esphome/core/component.h:4,
from src/esphome/components/api/api_message.h:3,
from src/esphome/components/api/basic_messages.h:3,
from src/esphome/components/api/basic_messages.cpp:1:
/root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/bits/uses_allocator.h:104:2: error: invalid preprocessing directive #en
#en
^
/root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/bits/uses_allocator.h:25:0: error: unterminated #ifndef
#ifndef _USES_ALLOCATOR_H
^
In file included from /root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/memory:77:0,
from src/esphome/core/helpers.h:6,
from src/esphome/components/api/util.h:3,
from src/esphome/components/api/api_message.h:4,
from src/esphome/components/api/command_messages.h:5,
from src/esphome/components/api/command_messages.cpp:1:
/root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/bits/uses_allocator.h:25:0: error: unterminated #ifndef
#ifndef _USES_ALLOCATOR_H
^
In file included from /root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/memory:77:0,
from src/esphome/core/helpers.h:6,
from src/esphome/components/api/util.h:3,
from src/esphome/components/api/api_message.h:4,
from src/esphome/components/api/basic_messages.h:3,
from src/esphome/components/api/basic_messages.cpp:1:
/root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/bits/uses_allocator.h:25:0: error: unterminated #ifndef
#ifndef _USES_ALLOCATOR_H
^
In file included from /root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/memory:77:0,
from src/esphome/core/helpers.h:6,
from src/esphome/components/light/light_color_values.h:3,
from src/esphome/components/light/light_effect.h:4,
from src/esphome/components/light/light_state.h:6,
from src/esphome/core/controller.h:11,
from src/esphome/components/api/api_server.h:4,
from src/esphome/components/api/api_server.cpp:3:
/root/.platformio/packages/toolchain-xtensa32/xtensa-esp32-elf/include/c++/5.2.0/bits/uses_allocator.h:25:0: error: unterminated #ifndef
#ifndef _USES_ALLOCATOR_H
^
*** [/data/werkbank/.pioenvs/werkbank/src/esphome/components/api/api_message.cpp.o] Error 1
*** [/data/werkbank/.pioenvs/werkbank/src/esphome/components/api/command_messages.cpp.o] Error 1
*** [/data/werkbank/.pioenvs/werkbank/src/esphome/components/api/basic_messages.cpp.o] Error 1
*** [/data/werkbank/.pioenvs/werkbank/src/esphome/components/api/api_server.cpp.o] Error 1
========================== [ERROR] Took 11.57 seconds ==========================
```

Component:  api_message.cpp.o

Configuration:

    substitutions:
    device_name: balkon_01
    api_pwd: !secret balkon_01_pwd
    
    esphome:
      name: ${device_name}
      platform: ESP32
      board: nodemcu-32s
    
    wifi:
      ssid: !secret wifi_ssid
      password: !secret wifi_password
    
    logger:
      baud_rate: 0
    
    #web_server:
    #  port: 80
    
    api:
      password: '${api_pwd}'
    
    ota:
      password: '${api_pwd}'
    
    switch:
      - platform: restart
        name: "${device_name}_restart"
    
    light:
      - platform: rgb
        name: '${device_name}_Light_01'
        red: output_red
        green: output_green
        blue: output_blue
        effects:
          - random:
          - strobe:
          - flicker:
    output:
      - platform: ledc
        id: output_red
        pin: GPIO16
      - platform: ledc
        id: output_green
        pin: GPIO17
      - platform: ledc
        id: output_blue
        pin: GPIO18
    
    i2c:
      sda: 21
      scl: 22
      scan: false
    
    sensor:
      - platform: uptime
        name: "${device_name}_Uptime"
      - platform: bh1750
        name: "${device_name}_Brightness"
        address: 0x23
        update_interval: 30s
        filters:
          - sliding_window_moving_average:
                window_size: 20
                send_every: 10
          - delta: 0.1
      - platform: bme280
        address: 0x76
        update_interval: 30s  
        temperature:
          name: "${device_name}_Temperature"
          filters:
            - offset: -1.5
            - delta: 0.1
            - sliding_window_moving_average:
                window_size: 20
                send_every: 10
          oversampling: 16x
        pressure:
          name: "${device_name}_Pressure"
          filters:
            - delta: 0.1
            - offset: -1000
            - sliding_window_moving_average:
                window_size: 20
                send_every: 10
        humidity:
          name: "${device_name}_Humidity"
          filters:
            - delta: 0.1
            - sliding_window_moving_average:
                window_size: 20
                send_every: 10
    
    
    # Infrared
    remote_receiver:
      pin: 
        number: GPIO26
        # Remote Receiver Signal starts with a HIGH value. Usually this means you have to invert the signal using 'inverted: True'
        inverted: yes
      dump: all

Buildlog:

    INFO Reading configuration...
    INFO Generating C++ source...
    INFO Compiling app...
    INFO Running:  platformio run -d /config/esphome/balkon_01
    Processing balkon_01 (platform: espressif32@1.6.0; framework: arduino; board: nodemcu-32s)
    --------------------------------------------------------------------------------
    Verbose mode can be enabled via `-v, --verbose` option
    CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/nodemcu-32s.html
    PLATFORM: Espressif 32 > NodeMCU-32S
    HARDWARE: ESP32 240MHz 320KB RAM (4MB Flash)
    DEBUG: CURRENT(esp-prog) EXTERNAL(esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
    Library Dependency Finder -> http://bit.ly/configure-pio-ldf
    LDF MODES: FINDER(chain) COMPATIBILITY(soft)
    Collected 30 compatible libraries
    Scanning dependencies...
    Dependency Graph
    |-- <AsyncTCP> 1.0.3
    |-- <ESPmDNS> 1.0
    |   |-- <WiFi> 1.0
    |-- <Preferences> 1.0
    |-- <Wire> 1.0.1
    |-- <WiFi> 1.0
    |-- <Update> 1.0
    Compiling /data/balkon_01/.pioenvs/balkon_01/src/esphome/components/api/api_message.cpp.o
    Compiling /data/balkon_01/.pioenvs/balkon_01/src/esphome/components/api/api_server.cpp.o
    Compiling /data/balkon_01/.pioenvs/balkon_01/src/esphome/components/api/basic_messages.cpp.o
    Compiling /data/balkon_01/.pioenvs/balkon_01/src/esphome/components/api/command_messages.cpp.o
    sh: 1: xtensa-esp32-elf-g++: not found
    *** [/data/balkon_01/.pioenvs/balkon_01/src/esphome/components/api/api_message.cpp.o] Error 127
    sh: 1: xtensa-esp32-elf-g++: not found
    *** [/data/balkon_01/.pioenvs/balkon_01/src/esphome/components/api/api_server.cpp.o] Error 127
    sh: 1: xtensa-esp32-elf-g++: not found
    *** [/data/balkon_01/.pioenvs/balkon_01/src/esphome/components/api/basic_messages.cpp.o] Error 127
    sh: 1: xtensa-esp32-elf-g++: not found
    *** [/data/balkon_01/.pioenvs/balkon_01/src/esphome/components/api/command_messages.cpp.o] Error 127
    ========================== [ERROR] Took 7.52 seconds ==========================

--- a/esphome/components/light/automation.py
+++ b/esphome/components/light/automation.py
@@ -155,6 +155,6 @@ def light_addressable_set_to_code(config, action_id, template_arg, args):
                                automation.maybe_simple_id({
                                    cv.Required(CONF_ID): cv.use_id(LightState),
                                }))
-def binary_sensor_is_on_to_code(config, condition_id, template_arg, args):
+def light_is_on_off_to_code(config, condition_id, template_arg, args):
     paren = yield cg.get_variable(config[CONF_ID])
     yield cg.new_Pvariable(condition_id, template_arg, paren)

--- a/esphome/config_validation.py
+++ b/esphome/config_validation.py
@@ -1175,7 +1175,6 @@ def validate_registry_entry(name, registry):
         if not isinstance(value, dict):
             raise Invalid(u"{} must consist of key-value mapping! Got {}"
                           u"".format(name.title(), value))
-        value = base_schema(value)
         key = next((x for x in value if x not in ignore_keys), None)
         if key is None:
             raise Invalid(u"Key missing from {}! Got {}".format(name, value))

--- a/tests/test1.yaml
+++ b/tests/test1.yaml
@@ -105,6 +105,9 @@ mqtt:
         - light.control:
             id: living_room_lights
             brightness: !lambda 'return id(living_room_lights).current_values.get_brightness() + 0.5;'
+        - light.dim_relative:
+            id: living_room_lights
+            relative_brightness: 5%
         - uart.write:
             id: uart0
             data: Hello World

